### ISSUE TITLE
Negative loss for segmentation on custom data

### ISSUE THREAD

**Issue by flomllr** [xu-ji/IIC#18]
Hi,
first of all, thank you for your great work.
I'm currently trying to set up the segmentation model for my custom data set. The training script is running, but I'm receiving a negative loss and the model is not really converging. (Accuracy still around 0.5 after 100 epochs.) Might that be an indication that something is wrong with my code?
Any help is greatly appreciated.
Thanks!
![image](https://user-images.githubusercontent.com/29047312/62525227-51681b00-b837-11e9-9188-3e547532990b.png)

**Comment by xu-ji** [comment]
Actually the loss is supposed to be negative, because mutual information is non-negative and the loss is negative mutual information. Also your accuracy increased from 48.5 to 54.7 between those two epochs, which is good. It looks like you have 13 classes? 54.7 is 7x better than random.

Do you have the plots for loss against epoch and accuracy against epoch? The main indicator of a problem is the loss not decreasing since the beginning of training.

**Comment by flomllr** [comment]
Hi, thanks for the clarification! Here are the plots for two different datasets:
![plots](https://user-images.githubusercontent.com/29047312/62683660-46d88d80-b9bf-11e9-9b0c-00e207670605.png)
![plots1](https://user-images.githubusercontent.com/29047312/62683672-49d37e00-b9bf-11e9-9ba5-2c4caddb0ca3.png)
So I think it is actually working.
Here is the visualization of one result:
Raw:
![1_img_0](https://user-images.githubusercontent.com/29047312/62684225-89e73080-b9c0-11e9-9ce6-25d86817ee1b.png)
Model output:
![1_reordered_preds_0](https://user-images.githubusercontent.com/29047312/62684260-953a5c00-b9c0-11e9-8915-3a1b2f9a5428.png)
Ground truth:
![1_targets_0](https://user-images.githubusercontent.com/29047312/62684268-9c616a00-b9c0-11e9-89b4-edfe8b109e99.png)

I'm particularly interested in extracting the road lines and sidewalks. Do you have an idea on how the performance could be improved for that task? (Maybe using less classes?)
Thanks a lot for your help!

**Comment by xu-ji** [comment]
Yes, that is quite a healthy loss curve. 

One thing to try, based on that visualisation, is decreasing the receptive field of the network. This would help with capturing finer details in the segmentation and getting the sizes of the segments more in proportion, as right now the segmentation is very coarse. 

There are two ways to change: either directly modifying receptive field by modifying the network architecture, or modifying effective receptive field by keeping the network the same and scaling down (for increased effective receptive field) or up (for decreased effective receptive field) the input images. The latter is similar to what we did for Potsdam. We cut each image into 15x15 squares and fed each square into the network instead of the whole original image. The receptive field (in pixels) of the network should be roughly tied to the size of the smallest detail you want to capture in its input images. 

Smaller receptive field might also help with the border effect issue (the green border).

It looks like you have 6 output classes from that image. But you are using 13?
